### PR TITLE
local storage door-config

### PR DESCRIPTION
**Pull request by FrankOelsch**
local storage door-config

### CODE REVIEW

File: src/pages/Config.js [hunk 1]
```diff
@@ -5,17 +5,35 @@ import Footer from "../components/Footer";
...
 export default function Config() {
   const canvasRef = useRef(null);
 
+  const [config, setConfig] = useState(
+    getFromLocalStorage("DoorConfig") ?? DoorConfig
+  );
+
   const messageWidthRef = useRef(null);
   const messageHeightRef = useRef(null);
   const messageRadiusRef = useRef(null);
```
Comment (ThomasF85) [inline src/pages/Config.js:35]: You are working with these refs to do direct dom manipulation (setting textContent). This is not how you are supposed to work in React. Please use states to describe the state of your components and use the state to change the appearance and text of your components.

File: src/pages/Config.js [hunk 2]
```diff
@@ -34,6 +52,10 @@ export default function Config() {
   let ctx = null;
   let canv = null;
 
+  useEffect(() => {
+    saveInLocalStorage("DoorConfig", config);
+  }, [config]);
```
Comment (ThomasF85) [inline src/pages/Config.js:57]: You are saving the new config into the localStorage even if the values do not pass your "validation" from the other useeffect. Do you want to save invalid values in localStorage?

File: src/pages/Config.js [hunk 3]
```diff
@@ -5,17 +5,35 @@ import Footer from "../components/Footer";
...
+}
+
+function getFromLocalStorage(key) {
+  try {
+    return JSON.parse(localStorage.getItem(key));
+  } catch (error) {
+    console.error(error.message);
+  }
+}
+
```
Comment (ThomasF85) [inline src/pages/Config.js:25]: You might want to consider using a localStorage hook instead. It's easier to use and it also handles cases like null, undefined values in localStorage.

**Review by ThomasF85 — CHANGES_REQUESTED** [review]
Please have a look at my comments.

File: src/pages/Config.js (hunk 3 above)
Comment (FrankOelsch) [reply src/pages/Config.js]: Have changed the suggested changes except the custom hook. Please give me your ok.

File: src/pages/Config.js [hunk 4]
```diff
@@ -4,17 +4,40 @@ import styled from "styled-components";
...
-  const messageWidthRef = useRef(null);
-  const messageHeightRef = useRef(null);
-  const messageRadiusRef = useRef(null);
+  const [config, setConfig] = useState(
+    getFromLocalStorage("DoorConfig") ?? DoorConfig
+  );
+
+  // const [config, setConfig] = useState(
+  //   useLocalStorage("DoorConfig", { DoorConfig })
+  // );
```
Comment (Kheirah) [inline src/pages/Config.js:36]: Try getting the hook to work before your next review =)

**Review by Kheirah — APPROVED** [review]
Well done!

File: src/pages/Config.js (hunk 4 above)
Comment (FrankOelsch) [reply src/pages/Config.js]: Now also useLocalStorage with help from our studies.

**Comment by FrankOelsch** [comment]
Now it works!

**Pull request merged**